src/nix/modules/service/docker-compose-service.nix

```diff
@@ -250,6 +273,8 @@ in
     inherit (config.service) container_name;
   } // lib.optionalAttrs (config.service.depends_on != []) {
     inherit (config.service) depends_on;
+  } // lib.optionalAttrs (config.service.healthcheck.test != null) {
+    inherit (config.service) healthcheck;
```

Comment:
This doesn't support disable yet. I suppose that could be added later.

Reply:
What do you suggest instead? We cannot add healthcheck options if it's not defined to avoid overriding healthcheck defined orginally in Dockerfile

Reply:
Yeah, it's a bit tricky, because we can't have hardcoded defaults. I think it can be solved with `options`, as opposed to `config`. `options.*.highestPrio` will let us figure out if the user defined a value.

If  you make `service.healthcheck` a submodule, you can do

```suggestion
  } // lib.optionalAttrs (options.service.healthcheck.highestPrio < 1500) {
    inherit (config.service) healthcheck;
```

which is a more general solution for that part of the problem.

It does make the sub-options a little inaccessible from this module level, so we'll need an internal option that gives us the filtered value we need.

```
    service.healthcheck = lib.mkOption {
      type = submodule ({ config, options, ... }: {
        options = {
          _out = mkOption {
            internal = true;
            default =
              optionalAttrs (options.test.highestPrio < 1500) { inherit (config) test; }
              // optionalAttrs #  ...
          };
          test = mkOption {
            # ...
```

```suggestion
  } // lib.optionalAttrs (options.service.healthcheck.highestPrio < 1500) {
    healthcheck = config.service.healthcheck._out;
```

Reply:
I've pushed some changes, take a look if it's what you mean